mypy/build.py

```diff
@@ -1272,7 +1272,8 @@ def __init__(self,
                  ancestor_for: 'State' = None,
                  root_source: bool = False,
                  ) -> None:
-        assert id or path or source is not None, "Neither id, path nor source given"
+        assert id is not None or path is not None or source is not None, \
+            "Neither id, path nor source given"
```

Comment:
I believe it is acting as a correlative conjunction thus "nor" is correct. The rule I was taught is "Either or, neither nor". But English is a very complex language and I won't pretend to be an expert in it.